### PR TITLE
[testlib] Add ssh_audit test to hpctestlist

### PR DESCRIPTION
**Pull request by victorusu**
The test downloads the ssh-audit tool from GitHub and checks if there is any configuration that is considered as a failure.

The test is compatible with all the job schedulers. Because it only checks the localhost, where the test script runs. So, it becomes the responsibility of ReFrame to spawn the test script in the desired systems.

### CODE REVIEW

File: hpctestlib/system/ssh/ssh_audit.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    `ssh-audit is a tool for ssh server & client configuration auditing.
+
+    The check consist on performing the basic ssh server config auditing
+    using the master version of https://github.com/jtesta/ssh-audit.
+    '''
+
+    executable = './ssh-audit.py'
+    executable_opts = ['-n', '-l', 'fail', 'localhost']
+    sourcesdir = 'https://github.com/jtesta/ssh-audit'
+    tags = {'system', 'ssh'}
```
Comment (vkarak) [inline hpctestlib/system/ssh/ssh_audit.py:23]: Should we add a security-related tag? Maybe just `security` or something more specific.

**Review by vkarak — APPROVED** [review]
Lgtm, just a minor comment.

**Comment by vkarak** [comment]
Also, have you checked the docs? Do these tests' docs appear correctly in the lib docs?

File: hpctestlib/system/ssh/ssh_audit.py (hunk 1 above)
Comment (victorusu) [reply hpctestlib/system/ssh/ssh_audit.py:23]: I don't know. I don't want to give the impression that this is a real security check. what do you think?

**Review by vkarak — APPROVED** [review]

**Pull request closed without merging**